components/scream/src/control/tests/dummy_grid.hpp

```diff
@@ -53,7 +53,11 @@ class DummyPhysicsGridRemapper : public AbstractRemapper<ScalarType,DeviceType>
     return src_layout;
   }
 
+// Cuda does not allow enclosing function of a lambda to have private or protected access
+#ifndef KOKKOS_ENABLE_CUDA
```

Comment:
I think @ambrad suggested we make this a macro:

```
// Clarify that a class member type is meant to be private but is
// marked public for Cuda visibility.
#ifdef KOKKOS_ENABLE_CUDA
# define PRIVATE_CUDA public
# define PROTECTED_CUDA public
#else
# define PRIVATE_CUDA private
# define PROTECTED_CUDA protected
#endif
```

Reply:
We can do that, if you want. Perhaps with clearer names, since reading 'PROTECTED_CUDA' makes me think it is indeed protected on cuda. Maybe something like 'KERNEL_CLASS_SCOPE' or something along that line. I'd like to avoid having `PROTECTED` or `PUBLIC` in the name, if those macros do not expand to `protected` or `public`.

Reply:
I'm ok with whatever name you think is good.

Reply:
LoL, careful what you wish for...

I will push a commit soon.